charms/knative-serving/src/manifests/KnativeServing.yaml.j2

```diff
@@ -28,9 +29,35 @@ spec:
       # Where local-gateway-namespace is the same as the knative-namespace and istio-namespace is where istio is deployed, in this particular
       # case we know the gateway namespace and the istio namespace will be the same
       local-gateway.{{ serving_namespace }}.knative-local-gateway: "knative-local-gateway.{{ gateway_namespace }}.svc.cluster.local"
+    {% endif %}
     # This is analogous to the config-domain configmap
+    # TODO: defining this gateway config did nothing.  Maybe the operator doesn't support it?
+    {% if ingress_class == 'gateway-api.ingress.networking.knative.dev' %}
```

Comment:
tbh I have no idea what I'm looking at here, what exactly is this API group and what is it used for?

Reply:
:smile: 

[this doc](https://knative.dev/docs/serving/setting-up-custom-ingress-gateway/) might help.  the `gateway` part of this configures the external and internal gateways that knative will use when creating VirtualService or HTTPRoute objects.  

by default, knative will use class Istio APIs to create everything.  But if we set `ingress-class: gateway-api.ingress.networking.knative.dev`, that turns on the new gateway APIs.  So what the jinja is doing here is:
* if ingress_class is set to gateway-api, define these gateway settings which are specific to gateway-api
* a little further down, if `ingress_class` is set to anything, then add the ingress_class entry in the configmap

Reply:
its hard to read and I'm happy to edit if you have suggestions.  what gave me trouble is that omitting the ingress_class means "use classic istio api", so then there's those odd looking `if not ingress_class` calls